providers.tf

```diff
@@ -0,0 +1,47 @@
+terraform {
+  required_version = ">= 0.12"
+}
+
+provider "aws" {
+  region = var.aws_region
+}
+
+data "aws_availability_zones" "available" {}
+
+# Not required: currently used in conjunction with using
+# icanhazip.com to determine local workstation external IP
+# to open EC2 Security Group access to the Kubernetes cluster.
+# See workstation-external-ip.tf for additional information.
+provider "http" {}
+
+provider "kubernetes" {
+  host                   = aws_eks_cluster.demo.endpoint
+  cluster_ca_certificate = base64decode(aws_eks_cluster.demo.certificate_authority[0].data)
+  exec {
```

Comment:
This is totally fine, but does introduce a dependency on the AWS CLI in order to execute properly.  Another option would be to pull the `kubeconfig` out of the `aws_eks_cluster.demo` resources and store it in a local file, and then reference that file here.

It's a little trickier with EKS because of the way AWS authenticates users and roles with EKS, so don't worry about refactoring it now.  More of an FYI on other options.